public/app/features/plugins/extensions/utils.tsx

```diff
@@ -159,3 +159,22 @@ function isRecord(value: unknown): value is Record<string | number | symbol, unk
 export function isReadOnlyProxy(value: unknown): boolean {
   return isRecord(value) && value[_isProxy] === true;
 }
+
+export function createExtensionLinkConfig<T extends object>(
+  config: Omit<PluginExtensionLinkConfig<T>, 'type'>
+): PluginExtensionLinkConfig {
+  const linkConfig = {
+    type: PluginExtensionTypes.link,
+    ...config,
+  };
+  assertIsLinkConfig(linkConfig);
+  return linkConfig;
+}
```

Comment:
Maybe it would make sense to just do this configuration inline (in `getExploreExtensions.tsx`) instead of the abstraction?

Reply:
hmm, can you elaborate a bit more on that? It is being used when registering the extension. Similar way as the plugin does it when registering via the `configureLinkExtension` function.